models/dojo.py

```diff
@@ -0,0 +1,277 @@
+            self.persons[fellow_id] = new_user
+
+            print("{0} {1} has been successfully added".format(self.persons[
+                fellow_id].rank, self.persons[fellow_id]))
+
+        else:
+            raise Exception('Person can only be a fellow or staff')
+
+        self.assign_person(new_user)
+
+    @staticmethod
+    def get_available_room_spaces(room_spaces):
+        available_room_spaces = []
+        for room, room_info in room_spaces.items():
+            if (len(room_info.occupants) < room_info.max_occupants):
+                available_room_spaces.append(room_info)
+        return available_room_spaces
+
+    def assign_person(self, person):
+        """The function randomly assigns a person to a room."""
+        available_office_spaces = self.get_available_room_spaces(
+            self.office_rooms)
+        if available_office_spaces == []:
+            print("There is currently no office space")
+        else:
+            assigned_office_space = random.choice(available_office_spaces)
+            assigned_office_space.occupants.append(person)
+            person.office_space_allocated = assigned_office_space.room_name
+            print("{0} has been allocated the office {1}".format(
+                person.first_name, assigned_office_space.room_name))
+
+        if person.wants_accomodation.upper() == "Y":
+            available_living_spaces = self.get_available_room_spaces(
+                self.living_rooms)
+            if available_living_spaces == []:
+                print("There is currently no living space")
+            else:
+                assigned_living_space = random.choice(available_living_spaces)
+                assigned_living_space.occupants.append(person)
+                person.living_space_allocated = assigned_living_space.room_name
+                print("{0} has been allocated the livingspace {1}".format(
+                    person.first_name, assigned_living_space.room_name))
+
+    def print_room(self, arg):
+        """It prints the names of occupantsstated room name."""
+        room_name = arg["<room_name>"]
+        if room_name in self.office_rooms:
+            office_room_info = self.office_rooms[room_name]
+            if office_room_info.occupants == []:
+                raise Exception("Office room is empty")
+            else:
+                print('Office room occupants --- {}'.format(office_room_info.occupants))
+
+        elif room_name in self.living_rooms:
+            living_room_info = self.living_rooms[room_name]
+            if living_room_info.occupants == []:
+                raise Exception("Living room is empty")
+            else:
+                print('Living room occupants --- {}'.format(living_room_info.occupants))
+        else:
+            raise Exception("The room has not been created")
+
+    def print_allocations(self, arg):
+        """The function prints a list of all
+         allocated rooms with their members."""
+        # pdb.set_trace()
```

Comment:
Before submitting a PR, it is good practice to go through the diff to make sure that your remove all debugging helpers.